realm/realm-library/src/main/java/io/realm/BaseRealm.java

```diff
@@ -98,6 +101,19 @@ public void onSchemaVersionChanged(long currentVersion) {
                                 }
                             }
                         }, true);
+        this.shouldCloseSharedRealm = true;
+        this.schema = new RealmSchema(this);
+    }
+
+    // Create a realm instance directly from a SharedRealm instance. This instance doesn't have the ownership of the
+    // given SharedRealm instance. The SharedRealm instance should not be closed when close() called.
+    BaseRealm(SharedRealm sharedRealm) {
```

Comment:
Please convert this constructor and the one above it into constructors that delegate to a single, third, constructor.